core/lib/spree/core/permalinks.rb

```diff
@@ -47,18 +47,14 @@ def generate_permalink
         "#{self.class.permalink_prefix}#{Array.new(self.class.permalink_length){rand(9)}.join}"
       end
 
-      def save_permalink(permalink_value=self.to_param)
+      def save_unique_permalink(permalink_value=self.to_param)
         self.with_lock do
           permalink_value ||= generate_permalink
-
           field = self.class.permalink_field
             # Do other links exist with this permalink?
             other = self.class.where("#{self.class.table_name}.#{field} LIKE ?", "#{permalink_value}%")
             if other.any?
-              # Find the existing permalink with the highest number, and increment that number.
-              # (If none of the existing permalinks have a number, this will evaluate to 1.)
-              number = other.map { |o| o.send(field)[/-(\d+)$/, 1].to_i }.max + 1
-              permalink_value += "-#{number.to_s}"
+              save_unique_permalink(permalink_value)
```

Comment:
1. Good catch, removing the duplicate permalink from the being passed in the method
2. Not fully sure how to address this one. Maybe we can pair on it? 

FYI made a new PR comparing against master (https://github.com/bonobos/solidus/pull/28) and closing this one.